File: src/com/github/virusmxa/things/subsetsum/AverageComparator.java
/**
 * Class AverageComparator compares the deviations from the mean.
 * 
 * @author Pavel Nikitin
 * License: GNU GPLv3
 * Modified: 2013/04/02
 */ 
package com.github.virusmxa.things.subsetsum;

import java.util.Comparator;

public class AverageComparator implements Comparator<Integer> 
{
    private int _average;
    
    public AverageComparator(int average) 
    {
        _average = average;
    }

    public int compare(Integer x, Integer y) 
    {
        return Math.abs(x - _average) - Math.abs(y - _average);
    }
}

File: src/AverageComparator.kt
/** Сравнение со средним арифметическим */

import java.util.*
import kotlin.math.abs

class AverageComparator(private val average: Int) : Comparator<Int> {
  override fun compare(x: Int, y: Int): Int {
    return abs(x - average) - abs(y - average)
  }
}